src/soundio/sounddevicenetwork.cpp

```diff
@@ -462,33 +465,34 @@ void SoundDeviceNetwork::callbackProcessClkRef() {
         }
     }
 
-    m_pSoundManager->readProcess();
+    m_pSoundManager->readProcess(framesPerBuffer);
 
     {
         ScopedTimer t("SoundDevicePortAudio::callbackProcess prepare %1",
                 m_deviceId.name);
-        m_pSoundManager->onDeviceOutputCallback(m_framesPerBuffer);
+        m_pSoundManager->onDeviceOutputCallback(framesPerBuffer);
     }
 
-    m_pSoundManager->writeProcess();
+    m_pSoundManager->writeProcess(framesPerBuffer);
 
-    m_pSoundManager->processUnderflowHappened();
+    m_pSoundManager->processUnderflowHappened(framesPerBuffer);
 
-    updateAudioLatencyUsage();
+    updateAudioLatencyUsage(framesPerBuffer);
 }
 
-void SoundDeviceNetwork::updateCallbackEntryToDacTime() {
+void SoundDeviceNetwork::updateCallbackEntryToDacTime(SINT framesPerBuffer) {
     m_clkRefTimer.start();
     qint64 currentTime = m_pNetworkStream->getInputStreamTimeUs();
-    m_targetTime += m_audioBufferTime.toIntegerMicros();
+    // This deadline for the next buffer in microseconds since the Unix epoch
+    m_targetTime += static_cast<qint64>(framesPerBuffer / m_dSampleRate * 1000000);
     double callbackEntrytoDacSecs = (m_targetTime - currentTime) / 1000000.0;
     callbackEntrytoDacSecs = math_max(callbackEntrytoDacSecs, 0.0001);
     VisualPlayPosition::setCallbackEntryToDacSecs(callbackEntrytoDacSecs, m_clkRefTimer);
```

Comment:
I understand that for rate syncing with the frequency of the network clock `framesPerBuffer` is enough. But to show the visual waveform, not only rate, but also latency is needed.
I don't understand this calculation. Shouldn't the waveform be in sync with the sound that the DJ hears (headphones or booth speakers) even if he's broadcasting a network stream?

Reply:
In this case the latency of the main DAC is lost. It would be an extra feature to compensate this. 

This calculation is only to compensate the jitter that happens due to the different cycles of the audio and video thread. 

At least this is unrelated to this PR. Do we want to file a bug for that?

Reply:
Yes, please create a bug for this. It would be great, if you could add some pointers to this report, how a possible fix could look like.
I agree, that this is not related to this PR.

Reply:
Here it is: https://github.com/mixxxdj/mixxx/issues/11219